src/assignment/views/AssignmentEdit.tsx

```diff
@@ -734,11 +738,11 @@ const AssignmentEdit: FunctionComponent<AssignmentEditProps> = ({
 		}
 
 		if (
-			!canEditAllAssignments ||
-			(assignment &&
-				!isUserAssignmentOwner(user, assignment) &&
-				!isUserAssignmentContributor(user, assignment) &&
-				!canEditAllAssignments)
+			// !canEditAllAssignments ||
+			assignment &&
+			!isUserAssignmentOwner(user, assignment) &&
+			!isUserAssignmentContributor(user, assignment) &&
+			!canEditAllAssignments
```

Comment:
Assignment is always defined, because it is the form state
You want to check for an id to be sure it is an exisiting assignment that has been fetched from the database
```suggestion
			!canEditAllAssignments &&
			assignment?.id &&
			!isUserAssignmentOwner(user, assignment) &&
			!isUserAssignmentContributor(user, assignment)
```

Reply:
assignment is possibly undefined somehow and typescript complains about it if i remove the check
<img width="1000" alt="image" src="https://github.com/viaacode/avo2-client/assets/113341869/55a53f41-5c46-4eca-94e6-380e25dc352a">